sentry_sdk/profiler.py

```diff
@@ -377,10 +377,23 @@ class _SleepScheduler(_ThreadScheduler):
 
     def run(self):
         # type: () -> None
+        last = time.perf_counter()
+
         while True:
+            # some time may have elapsed since the last time
+            # we sampled, so we need to account for that and
+            # not sleep for too long
+            now = time.perf_counter()
+            elapsed = max(now - last, 0)
+
+            if elapsed < self._interval:
```

Comment:
I'm not sure why you test this before you sleep. Shouldn't we sleep all the time at least a little bit, just the remaining amount of time until `self._interval` passed?

Reply:
If the elapsed time is greater than `self._interval`, then we'd want to sample as soon as possible otherwise the time between the 2 samples will be much larger than expected. This is because other threads could be running without giving this thread a chance to execute.